Sources/HTTP/Responder/HTTPServer.swift

```diff
@@ -236,10 +236,26 @@ private final class HTTPServerHandler<R>: ChannelInboundHandler where R: HTTPSer
         // a valid request
         var reshead = res.head
         reshead.headers.add(name: "date", value: RFC1123DateCache.shared.currentTimestamp())
+
         if let server = serverHeader {
             reshead.headers.add(name: "server", value: server)
         }
 
+        // add 'Connection' header if needed
+        let connectionHeaders = reshead.headers[canonicalForm: "connection"].map { $0.lowercased() }
+
+        if !connectionHeaders.contains("keep-alive") && !connectionHeaders.contains("close") {
```

Comment:
I'd replace the whole body of this commit with

    reshead.headers.add(name: "Connection", value: reqhead.isKeepAlive ? "keep-alive" : "close")

NIO's [example](https://github.com/apple/swift-nio/blob/master/Sources/NIOHTTP1Server/main.swift#L43-L53) make this slightly more complicated to not even set the header if we match the default of the protocol but that's probably unnecessary.

Reply:
But with the `connectionHeaders` we check if the response contains any `connection` header so we don’t duplicate it. Shouldn’t we keep it?

This said, agree on simplifying the rest of the code with just:

```
reshead.headers.add(name: "Connection", value: reqhead.isKeepAlive ? "keep-alive" : "close")
```

Reply:
@enricenrich apologies, totally missed that. That dupe checking looks good then :).